src/xercesc/validators/common/DFAContentModel.cpp

```diff
@@ -1364,14 +1372,27 @@ unsigned int DFAContentModel::countLeafNodes(ContentSpecNode* const curNode)
         if(nLoopCount!=0)
         {
             count += countLeafNodes(cursor);
-            for(unsigned int i=0;i<nLoopCount;i++)
-                count += countLeafNodes(rightNode);
+            const unsigned int countRight = countLeafNodes(rightNode);
+            // Avoid integer overflow in below multiplication
+            if (countRight > std::numeric_limits<unsigned int>::max() / nLoopCount)
```

Comment:
In all of the following changes, just want to check as above that the limit check should use `XMLSize_t` rather than `unsigned int`?

And would it also be possible to put brackets around all expressions being compared for clarity?

Reply:
cf above comments regarding data type

brackets added